packages/@n8n/permissions/test/hasScope.test.ts

```diff
@@ -0,0 +1,124 @@
+import { hasScope } from '@/hasScope';
+import type { AllScopes } from '@/types';
+
+const ownerPermissions: AllScopes[] = [
+	'workflow:create',
+	'workflow:read',
+	'workflow:update',
+	'workflow:delete',
+	'workflow:list',
+	'user:create',
+	'user:read',
+	'user:update',
+	'user:delete',
+	'user:list',
+	'credential:create',
+	'credential:read',
+	'credential:update',
+	'credential:delete',
+	'credential:list',
+	'variable:create',
+	'variable:read',
+	'variable:update',
+	'variable:delete',
+	'variable:list',
+];
+const memberPermissions: AllScopes[] = ['user:list', 'variable:list', 'variable:read'];
+
+describe('hasScope', () => {
+	test('should work with a single permission on both modes with only global scopes', () => {
+		expect(
+			hasScope(
+				'user:list',
+				{
+					global: memberPermissions,
+				},
+				{ mode: 'oneOf' },
+			),
+		).toBe(true);
+
+		expect(
+			hasScope(
+				'user:list',
+				{
+					global: memberPermissions,
+				},
+				{ mode: 'allOf' },
+			),
+		).toBe(true);
+
+		expect(
+			hasScope(
+				'workflow:read',
+				{
+					global: memberPermissions,
+				},
+				{ mode: 'oneOf' },
+			),
+		).toBe(false);
+
+		expect(
+			hasScope(
+				'workflow:read',
+				{
+					global: memberPermissions,
+				},
+				{ mode: 'allOf' },
+			),
+		).toBe(false);
+	});
+
+	test('should work with oneOf mode', () => {
```

Comment:
Here we're testing by explicitly passing the default, but since it's the default it is likely to be called without it. Should we test also by relying on the default?